src/main/java/spireMapOverhaul/zones/gremlinTown/powers/MushroomSoupPower.java

```diff
@@ -0,0 +1,34 @@
+package spireMapOverhaul.zones.gremlinTown.powers;
+
+import com.evacipated.cardcrawl.mod.stslib.actions.tempHp.AddTemporaryHPAction;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.localization.PowerStrings;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.abstracts.AbstractSMOPower;
+import spireMapOverhaul.zones.gremlinTown.GremlinTown;
+
+import static spireMapOverhaul.util.Wiz.*;
+
+public class MushroomSoupPower extends AbstractSMOPower {
+    public static final String POWER_ID = SpireAnniversary6Mod.makeID("MushroomSoup");
+    private static final PowerStrings powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
+    public static final String NAME = powerStrings.NAME;
+    public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
+
+    public MushroomSoupPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, GremlinTown.ID, PowerType.BUFF,false, owner, amount);
+        canGoNegative = false;
+    }
+
+    @Override
+    public void updateDescription() {
+        description = DESCRIPTIONS[0].replace("{0}", Integer.toString(amount));
+    }
+
+    @Override
+    public void atEndOfTurn(boolean isPlayer) {
+        atb(new AddTemporaryHPAction(adp(), adp(), amount));
```

Comment:
```suggestion
        atb(new AddTemporaryHPAction(owner, owner, amount));
```
Maybe it's only supposed to be on the player, but you have a owner, so use it